assets/js/modules/tagmanager/hooks/useExistingTagEffect.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Tag Manager useExistingTag custom hook.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_TAGMANAGER } from '../datastore/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function useExistingTagEffect() {
```

Comment:
Can we add unit tests for this hook? You can probably use the previous tests except the removed selectors :)

<img width="1299" alt="Screenshot 2022-03-01 at 3 52 18 PM" src="https://user-images.githubusercontent.com/24408261/156151713-40591ab1-9932-4712-8bea-2139c0ec31ee.png">

https://github.com/google/site-kit-wp/pull/4750/files